### PR TITLE
Fix bug with collateral selection for KeyWallet, Add `SignMultiple` example for KW

### PR DESCRIPTION
**Pull request by errfrom**
Closes https://github.com/Plutonomicon/cardano-transaction-lib/issues/668

I was able to reproduce the bug by running `Examples.KeyWallet.SignMultiple` and signing the transaction using a wallet with two ADA-only utxos with values of 5 and 10 ADA. The problem was the utxo with the lowest ADA value was selected as the collateral input for both transactions, and it was also spent by the first one, causing the second transaction to fail with the following error message: 

```
"SubmitFail":[{"badInputs":[{"txId":"0eb135218e6f93ac35e420fde373153abeeab5d218fdec66ac7ad605189a8b6e","index":2}]}]
```

In this PR I provided a basic solution, which is to select a collateral from the set of available utxos. 

### Pre-review checklist

- [x] All code has been formatted using our config (`make format` for Purescript, `nixpkgs-fmt` for Nix)
- [x] All Purescript imports are explicit, including constructors
- [x] **All** existing examples have been run locally against a fully-synced testnet node
- [x] The integration and unit tests have been run (`npm run test`) locally
- [x] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`)

### CODE REVIEW

**Comment by ngua** [comment]
Agreed @jy14898 this should be against `develop`. Especially since there are issues with light wallets since the hardfork, users might want to use the key wallet instead

**Review by ngua — COMMENTED** [review]
Just a couple of things, otherwise looks good

File: src/BalanceTx.purs [hunk 1]
```diff
@@ -413,6 +414,44 @@ _redeemersTxIns = lens' \(UnattachedUnbalancedTx rec@{ redeemersTxIns }) ->
...
+  runExceptT do
+    wallet <- asks _.wallet
+    mCollateral <- ExceptT $ selectCollateral wallet
+    traceM ("collateral: " <> show mCollateral)
+    case (mCollateral /\ wallet) of
+      Nothing /\ _ ->
+        pure $ transaction
+      Just collateral /\ (Just (KeyWallet _)) -> do
+        let collTx = addTxCollateral collateral transaction
+        pure collTx -- TODO:
```
Comment (klntsky) [inline src/BalanceTx.purs:433]: Please factor out `pure`.

File: src/BalanceTx.purs [hunk 2]
```diff
@@ -413,6 +414,44 @@ _redeemersTxIns = lens' \(UnattachedUnbalancedTx rec@{ redeemersTxIns }) ->
...
+--------------------------------------------------------------------------------
+
+setCollateral
+  :: Transaction -> Utxo -> QueryM (Either BalanceTxError Transaction)
+setCollateral transaction utxos =
+  runExceptT do
+    wallet <- asks _.wallet
+    mCollateral <- ExceptT $ selectCollateral wallet
+    traceM ("collateral: " <> show mCollateral)
+    case (mCollateral /\ wallet) of
```
Comment (klntsky) [inline src/BalanceTx.purs:428]: Unnecessary parentheses

File: src/BalanceTx.purs [hunk 3]
```diff
@@ -413,6 +414,44 @@ _redeemersTxIns = lens' \(UnattachedUnbalancedTx rec@{ redeemersTxIns }) ->
...
+      Nothing /\ _ ->
+        pure $ transaction
+      Just collateral /\ (Just (KeyWallet _)) -> do
```
Comment (klntsky) [inline src/BalanceTx.purs:431]: Unnecessary parentheses

File: src/BalanceTx.purs [hunk 4]
```diff
@@ -413,6 +414,44 @@ _redeemersTxIns = lens' \(UnattachedUnbalancedTx rec@{ redeemersTxIns }) ->
   redeemersTxIns /\
     \rdmrs -> UnattachedUnbalancedTx rec { redeemersTxIns = rdmrs }
 
+--------------------------------------------------------------------------------
+-- Setting collateral, collateral return, total collateral
+--------------------------------------------------------------------------------
+
+setCollateral
+  :: Transaction -> Utxo -> QueryM (Either BalanceTxError Transaction)
+setCollateral transaction utxos =
```
Comment (klntsky) [inline src/BalanceTx.purs:423]: The variable name is in plural form, but the type is in singular, could you please change either?

File: src/BalanceTx.purs (hunk 1 above)
Comment (klntsky) [inline src/BalanceTx.purs:433]: TODO what?

**Review by klntsky — CHANGES_REQUESTED** [review]

**Comment by errfrom** [comment]
This PR with `develop` as base branch: https://github.com/Plutonomicon/cardano-transaction-lib/pull/722

**Pull request closed without merging**